src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,68 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        // Try to resolve implementation for the given interface
+        Class<?> implClass = findImplementation(interfaceClazz);
+
+        // Check if class is annotated with @Component
+        if (!implClass.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + implClass.getName()
+                    + " is not marked as @Component");
+        }
+
+        // Check if an instance already exists
+        if (instances.containsKey(implClass)) {
+            return instances.get(implClass);
+        }
+
+        // Create a new instance and inject dependencies
+        Object instance = createInstance(implClass);
+        instances.put(implClass, instance);
+        return instance;
+    }
```

Comment:
Make Interface Implementations map a class field. You can fill it in using `Map.of()`. This will help to avoid using the `if` statements in the `findImplementation` method for determining the implementation class.